moments/Parsing.py

```diff
@@ -0,0 +1,1119 @@
+    ancestral alleles are read from it. Here also, sites are skipped when they
+    lack a valid ancestral allele. This behavior is modulated by 
+    `allow_low_confidence`. When True, sites assigned ancestral states in
+    lower-case (e.g. 'a') are retained. Otherwise such sites are skipped.
+
+    When `allow_multiallelic` is False and the ancestral allele is not 
+    represented as either the reference or alternate allele at a site, that 
+    site is skipped automatically. 
+
+    The parameter `filters` allows a number of criteria to be imposed on lines,
+    so that they can be filtered by quality or annotation. `filters` should be
+    a flat dictionary mapping strings to appropriately typed objects- the 
+    following are valid key-value pairs:
+    'QUAL' should map to a single number, imposing a minimum on the ``QUAL``
+        field. 
+    'FILTER' should map to a str or set/tuple/list of strings. ``FILTER``
+        must match the value (or an element thereof, if it is iterable)
+        to be parsed.
+    'FORMAT/FIELD' e.g. 'FORMAT/GQ' should map to a single number, or in
+        applicable cases to a str/set of strings. When a number, specifies a 
+        minimum value for a field. The typing of values is not explictly 
+        checked against the expected type for a field- e.g. if a string is 
+        given for the numeric field 'GQ', no explicit warning is raised- 
+        although an error will tend to be thrown once parsing begins.
+    'SAMPLE/FIELD' e.g. 'SAMPLE/GQ' imposes filters at the sample level. 
+        'FIELD' should correspond to an entry in the ``FORMAT`` column of the 
+        VCF. Typing is the same as in 'FORMAT'.
+    In general, when quantities are absent in a given line, the line is not 
+    skipped but a one-time alert message is raised. Missing values ('.') also
+    do not cause lines to be skipped. 
+
+    :param vcf_file: Pathname of the VCF file to parse. The file may be 
+        gzipped/bgzipped or uncompressed.
+    :type vcf_file: str
+    :param pop_file: Pathname of a whitespace-separated file mapping samples
+        to populations with the format SAMPLE POPULATION. Sample names must
+        be unique and there should be one of them on each line (default None 
+        combines all samples into a single population 'ALL'). Samples not 
+        listed here are ignored.
+    :type pop_file: str, optional
+    :param pop_mapping: Optional dictionary (default None) mapping population
+        IDs to lists of VCF sample IDs. Mutually exclusive with `pop_file`.
+    :type pop_mapping: dict, optional
+    :param bed_file: Pathname of a BED file defining the regions within which 
+        to parse; useful for applying masks to exclude difficult-to-call or 
+        functionally constrained regions (default None). 
+    :type bed_file: str, optional
+    :param interval: 2-tuple or 2-list specifying the (inclusive, 0-indexed) 
```

Comment:
VCF files are 1-indexed, no? We would never see a site at position 0, for example, or else that is an invalid VCF file.